sbin/prepareWorkspace.sh

```diff
@@ -324,6 +324,9 @@ checkingAndDownloadingAlsa() {
         # Use /tmp for alpine in preference to $HOME as Alpine fails gpg operation if PWD > 83 characters
         # Alpine also cannot create ~/.gpg-temp within a docker context
         GNUPGHOME="$(mktemp -d /tmp/.gpg-temp.XXXXXX)"
+    elif [ "${BUILD_CONFIG[OS_KERNEL_NAME]}" == "linux" ] && [ "${BUILD_CONFIG[OS_ARCHITECTURE]}" == "riscv64" ] && [ "$(pwd | wc -c)" -gt 83 ]; then
```

Comment:
Should we instead take this path if `GNUPGHOME` is not defined?

Reply:
not sure, we don't do in the build-farm path either ?

Reply:
I think the intention is to create it in the WORKSPACE_DIR

Reply:
> not sure, we don't do in the build-farm path either ?

We should add it there as well then. Should we move that specific piece of code to a separate bash script that we include from all relevant places? That sounds easier than copy/pasting it all around.

Reply:
@sxa I think you're more experienced with GPG than me, what's your view on honouring GNUPGHOME if it's already set in the environment? eg.by a "user"

Reply:
As long as we're ok with a user potentially shooting themselves in the foot by setting it something >83 characters and having us honour that. I thin it's a subtle enough case that I don't mind either way (I'd probably err on the side of not doing the check as it's less code but 🤷🏻 - I'll leave it to you. It's an edge case that won't affect our build process unless we chose to define it on our machines in jenkins or something similar.

Reply:
I'm also keen on less code, so will leave that as is for the moment